Realm/ObjectServerTests/RLMObjectServerTests.mm

```diff
@@ -1276,597 +1276,6 @@ - (void)testDownloadWhileOpeningRealm {
-    RLMNotificationToken *token = [r addNotificationBlock:^(RLMResults *results,
-                                                            __unused RLMCollectionChange *change,
-                                                            __unused NSError *error) {
-        if (results.count == 0) {
-            return;
-        }
-
-        RLMSyncPermissionOffer *permissionOffer = results[0];
-        if (permissionOffer.statusCode) {
-            XCTAssertEqual(permissionOffer.status, RLMSyncManagementObjectStatusError);
-            XCTAssertNil(permissionOffer.realmUrl);
-
-            [expectation fulfill];
-        }
-    }];
-
-    [self workAroundCoreIssue2724];
-    [managementRealm transactionWithBlock:^{
-        [managementRealm addObject:permissionOfferResponse];
-    } error:&error];
-    XCTAssertNil(error, @"Error when writing permission offer response object: %@", error);
-
-    [self waitForExpectationsWithTimeout:2.0 handler:nil];
-
-    [token stop];
-}
-
-/// Failed to process a permission offer response object due to `token` does not exist
-- (void)testPermissionOfferResponseTokenNotExist {
-    RLMSyncUser *user = [self logInUserForCredentials:[RLMObjectServerTests basicCredentialsWithName:NSStringFromSelector(_cmd)
-                                                                                            register:self.isParent]
-                                               server:[RLMObjectServerTests authServerURL]];
-
-    NSString *fakeToken = @"00000000000000000000000000000000:00000000-0000-0000-0000-000000000000";
-    RLMSyncPermissionOfferResponse *permissionOfferResponse = [RLMSyncPermissionOfferResponse permissionOfferResponseWithToken:fakeToken];
-
-    NSError *error = nil;
-    RLMRealm *managementRealm = [user managementRealmWithError:&error];
-    XCTAssertNotNil(managementRealm);
-    XCTAssertNil(error, @"Error when opening management Realm: %@", error);
-
-    XCTestExpectation *expectation = [self expectationWithDescription:@"A new permission offer response will be processed by the server"];
-    RLMResults *r = [RLMSyncPermissionOfferResponse objectsInRealm:managementRealm where:@"id = %@", permissionOfferResponse.id];
-    RLMNotificationToken *token = [r addNotificationBlock:^(RLMResults *results,
-                                                            __unused RLMCollectionChange *change,
-                                                            __unused NSError *error) {
-        if (results.count == 0) {
-            return;
-        }
-
-        RLMSyncPermissionOffer *permissionOffer = results[0];
-        if (permissionOffer.statusCode) {
-            XCTAssertEqual(permissionOffer.status, RLMSyncManagementObjectStatusError);
-            XCTAssertNil(permissionOffer.realmUrl);
-
-            [expectation fulfill];
-        }
-    }];
-
-    [self workAroundCoreIssue2724];
```

Comment:
The workarounds were removed because the tests that utilized the workaround were all deleted.